src/cluster_legacy.c

```diff
@@ -3045,7 +3045,11 @@ static void clusterProcessModulePacket(clusterMsgModule *module_data, clusterNod
     uint32_t len = ntohl(module_data->len);
     uint8_t type = module_data->type;
     unsigned char *payload = module_data->bulk_data;
-    moduleCallClusterReceivers(sender->name, module_id, type, payload, len);
+
+    /* Ensure sender name is properly truncated to 40 characters */
+    sds truncated_name = sdsnewlen(sender->name, 40);
+    moduleCallClusterReceivers(truncated_name, module_id, type, payload, len);
+    sdsfree(truncated_name);
```

Comment:
This is not truncating per se. This is more like adding a `\0` at the end (null terminated string). You could just name it as sender_id.

```suggestion
    sds sender_name = sdsnewlen(sender->name, 40);
    moduleCallClusterReceivers(sender_name, module_id, type, payload, len);
    sdsfree(sender_name);
```